processing/src/main/java/io/druid/query/ColumnSelectorPlus.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query;
+
+import io.druid.query.dimension.ColumnSelectorStrategy;
+import io.druid.segment.ColumnValueSelector;
+
+/**
+ * A grouping of various related objects used during query processing for a single dimension, used for convenience.
+ *
+ * Each ColumnSelectorPlus is associated with a single dimension.
+ */
+public class ColumnSelectorPlus<ColumnSelectorStrategyClass extends ColumnSelectorStrategy>
```

Comment:
I'm wondering if we're going to take a performance hit from the approach where we're calling `plus.getColumnSelectorStrategy()` and then passing `plus.getSelector()` to various methods on the strategy. It may be more performant to have the strategy be an adapter rather than a strategy (i.e. it "owns" the selector and presents an interface adapted to the query engine's needs).

But if the benchmarks look fine, there's no reason to make this adjustment. So let's do the benchmarks first.